db/1.9/modules/ROOT/pages/database-connector-query-single.adoc

```diff
@@ -0,0 +1,204 @@
+* *Add Maven dependency* +
+Install a Maven dependency to add to the project.
++
+[start=10]
+. In the *Connection* section, set the following parameters: +
++
+* *Host* +
+`localhost`
+* *Port* +
+`3306`
+* *User* +
+`root`
+* *Password* +
+`mysql`
+* *Database* +
+`sys`
++
+[start=11]
+. On the *Advanced* tab, set the *Pooling profile* field to *Edit inline*.
+. Set the *Max pool size* field to `5`.
+. Click *Test Connection* to confirm that Mule can connect to the database.
+. Click *OK*.
+
+The following screenshot shows the HTTP Listener global element configuration in Studio:
+
+.HTTP Listener configuration
+image::database-querysingle-example-1.png[HTTP Listener configuration]
+
+The following screenshots show the Database global element configuration in Studio:
+
+.Database Config General cofiguration
+image::database-querysingle-example-2.png[Database Config General configuration setting Host, Port, User, Password Database values parameters]
+
+== Create, Run and Test the Mule Application
+After you configure the MySQL database connection, create, run and test the Mule application:
+
+=== Configure the HTTP Listener and Set Variable Component
+
+To create the Mule flow:
+
+. In the *Mule Palette* view, select the HTTP *Listener* source and drag it on to the canvas. +
+The source initiates the flow by listening for incoming HTTP message attributes.
+. In the *Connector configuration* field, select `HTTP_Listener_config` global configuration.
+. Set the *Path* field to `/select/book/{maxId}`. +
+The `maxId` value parameter indicates how many books information you will retrieve and list from the database. You can increment this value number regardless of the number of maximum connections you configured in your database connection pool.
+. In the *Mime Type* tab, set the *Mime Type* field to `application/json`.
+. In the *Advanced* tab, set the *Allowed methods* field to `GET`.
+. Drag a *Set Variable* component to the right of the *Listener* source. +
+This component creates a new variable to save the database results that will be obtained from the *Single Query* operation.
+. Set the *Name* field to `bookCollection` and the *Value* field to `#[[]]`.
+
+.Set Variable configuration
+image::database-querysingle-example-3.png[Set Variable configuration]
+
+=== Configure the For Each Component, the Single Query Operation and Set Payload Component
+
+Continue creating the Mule application using a *For Each* component that iterates over the number of books requested in the *HTTP Listener*, retrieves the books information using the *Single Query* operation, save the results in a payload variable:
+
+. Drag a *For Each* component to the right of *Set Variable*. +
+. Set the *Collection* field to `#[1 to attributes.uriParams.maxId]`. +
+This expression iterates the collection of books in the table, from the first book (`1`) to the number of books set in the `maxId` parameter when performing the HTTP request `/select/book/{maxId}`.
```

Comment:
bold "maxId" and use *Max Id* if trying to match field names in Studio
Add a comma after "maxId parameter."